cypress-CI

Philaktet opened the pull request.

frendly commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,27 @@
+          node-version: 22.17.1
+
+      - name: Install dependencies
+        working-directory: ./E2E_tests
```
тут директория повторяется, мб вынести в переменную

Philaktet committed:
Update cypress-tests.yml

Добавил переменную WORKING_DIR: ./E2E_tests, 
урезал репорт отчета о тестах (--quiet --reporter minimal)

Philaktet committed:
TEST: minimal error-only output

Philaktet committed:
Merge branch 'cypress-CI' of github.com:Philaktet/QA_works into cypress-CI

Philaktet committed:
проба изменения фильтра

Philaktet committed:
1

Philaktet committed:
добавил cross-env

Philaktet committed:
попытка2

Philaktet committed:
попытка3

Philaktet committed:
негдобавил ошибку в тест поиска

Philaktet committed:
добавил ошибку в 1й тест -wkipedia.org

Philaktet committed:
попытка 4, ошибка 1го теста.

Philaktet committed:
возврат к предыдущей попытке

Philaktet committed:
возврат

Philaktet commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,27 @@
+
+      - name: Run Cypress tests
+        working-directory: ./E2E_tests
+        run: yarn test
```
Узнал что для публичного репозитория не нужно сохранять лимиты (их нет), но всё равно настроил работу на 5 секунд с минимальным выводом. Правда, он пропустил ошибку и сделал ✅ All tests passed...но с этим разберусь завтра до нашей итерации. Или на ней.

frendly replied:
о, круто!

Philaktet replied:
Настроил, можно проверять.

Philaktet committed:
подпись ошибки в тесте

Philaktet committed:
обновил йарнлок

frendly commented on E2E_tests/scripts/test-runner.js:
```diff
@@ -0,0 +1,17 @@
+}`;
+
+exec(command, (error, stdout) => {
+  const errors = stdout.match(/(✖|error:|fail:).*/gi);
```
видимо потому что используются недопустимые флаги
```
// E2E_tests/scripts/test-runner.js
console.log("stdout", stdout);

yarn test

stdout You passed the --ci-build-id, --group, --tag, --parallel, or --auto-cancel-after-failures flag without also passing the --record flag.

The --ci-build-id flag you passed was: local
The --parallel flag you passed was: true
The --auto-cancel-after-failures flag you passed was: undefined

These flags can only be used when recording to Cypress Cloud.

https://on.cypress.io/record-params-without-recording
```

Philaktet replied:
Спасибо, поправил.

frendly commented on E2E_tests/yarn.lock:
```diff
@@ -2,18 +2,18 @@
 # yarn lockfile v1
 
 
-"@cypress/request@^3.0.8":
```
в ПР никакие новые зависимости через yarn не устанавливали, но lock файл обновляется, почему?

Philaktet replied:
Не знаю. Надо будет почитать.

frendly commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,30 @@
+
+      - name: Install dependencies
+        working-directory: ${{ env.WORKING_DIR }}
+        run: yarn install --frozen-lockfile --prefer-offline  
```
а что такое prefer-offline?

Philaktet replied:
Флаг, указывающий смотреть на кеш, в первую очередь.

frendly commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,30 @@
+jobs:
+  cypress-run:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10 
```
вроде все тесты проходят быстро, а тут лимит в 10 минут, мб уменьшить?

Philaktet replied:
Убрал вообще таймаут.

frendly commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,30 @@
+    timeout-minutes: 10 
+    env:
+      WORKING_DIR: ./E2E_tests
+      CYPRESS_NO_COMMAND_LOG: 1  
```
определяем env переменные, но ниже они нигде не используются

Philaktet replied:
<img width="771" height="404" alt="image" src="https://github.com/user-attachments/assets/4f09b381-f70f-4453-a602-45966ae19685" />
По твоей просьбе переменная введена. Я не знаю пока зачем.  Зачем?)

frendly commented on .github/workflows/cypress-tests.yml:
```diff
@@ -0,0 +1,30 @@
+      - name: Setup Node.js (with cache)
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.17.1
```
как вариант, можно попробовать лочить мажорную lts версию
```suggestion
          node-version: 22
```

Philaktet replied:
А надо? Во-первых, путаница. Во-вторых, только что вписал .nvmrc-файл и поправил реадми.  Объяснишь зачем указывать иную версию?

Philaktet committed:
Merge branch 'main' into cypress-CI

Philaktet committed:
Add Acton-page button (cypress-tests.yml), change requare-->import\export (several files), upd test-runner.js, dell non-working flags

Philaktet committed:
Merge branch 'cypress-CI' of github.com:Philaktet/QA_works into cypress-CI

Philaktet committed:
Workin testrunner and tests√

Philaktet committed:
Merge branch 'main' into cypress-CI
